tasks/vagrant.rb

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env ruby
+require 'json'
+require 'net/http'
+require 'yaml'
+require 'puppet_litmus'
+require 'fileutils'
+
+@supported_platforms = {
+  'ubuntu14.04' => 'ubuntu/trusty64',
+}
+
+def run_local_command(command, wd = Dir.pwd)
+  stdout, stderr, status = Open3.capture3(command, chdir: wd)
+  error_message = "Attempted to run\ncommand:'#{command}'\nstdout:#{stdout}\nstderr:#{stderr}"
+  raise error_message unless status.to_i.zero?
+  stdout
+end
+
+def platform_uses_ssh(platform)
+  uses_ssh = if platform !~ %r{win-}
+               true
+             else
+               false
+             end
+  uses_ssh
+end
+
+def generate_vagrantfile(file_path, platform)
+  image = @supported_platforms[platform]
+  raise "Platform '#{platform}' not supported" unless image
+  vf = "Vagrant.configure(\"2\") do |config|\n"
+  vf << "  config.vm.box = '#{image}'\n"
+  vf << "  config.vm.boot_timeout = 600\n"
+  # fix username
+  vf << "  config.ssh.username = 'vagrant'\n"
+  vf << "  config.ssh.password = 'vagrant'\n"
+  vf << "  config.ssh.insert_key = false\n"
+  vf << "  config.vm.provision \"shell\", inline: <<-SHELL\n"
+  vf << "    sed -i 's/ChallengeResponseAuthentication no/ChallengeResponseAuthentication yes/g' /etc/ssh/sshd_config\n"
+  vf << "    sed -i 's/PermitRootLogin without-password/PermitRootLogin yes/g' /etc/ssh/sshd_config\n"
+  vf << "    service ssh restart\n"
+  # set root password
+  vf << "    echo 'root:vagrant'|chpasswd\n"
+  vf << "  SHELL\n"
+  vf << "end\n"
+  File.open(file_path, 'w') do |f|
+    f.write(vf)
+  end
+end
+
+def provision(platform, inventory_location)
+  include PuppetLitmus
+  # TODO: check for ports
+  node_name = 'localhost:2222'
+  vagrant_location = File.join(inventory_location, '.vagrant', node_name)
+  FileUtils.mkdir_p vagrant_location
+  generate_vagrantfile(File.join(vagrant_location, 'Vagrantfile'), platform)
+  command = 'vagrant up --provider virtualbox'
+  run_local_command(command, vagrant_location)
+  vm_id = File.read(File.join(vagrant_location, '.vagrant', 'machines', 'default', 'virtualbox', 'index_uuid'))
+  if platform_uses_ssh(platform)
+    node = { 'name' => node_name,
+             'config' => { 'transport' => 'ssh', 'ssh' => { 'user' => 'root', 'host' => 'localhost', 'password' => 'vagrant', 'host-key-check' => false, 'port' => 2222 } },
```

Comment:
```suggestion
             'config' => { 'transport' => 'ssh', 'ssh' => { 'user' => 'vagrant', 'host' => 'localhost', 'password' => 'vagrant', 'host-key-check' => false, 'port' => 2222 } },
```
then you dont need to worry about root ?